### PR TITLE
fix(utils): export common utility functions and constants in react-core

### PR DESCRIPTION
**Pull request by priley86**
**What**:
A minor minor addition to react-core to help out #1227 

* exports some common util functions/constants and convert them to typescript

**Additional issues**:
none

### CODE REVIEW

File: packages/patternfly-4/react-core/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export * from './components';
 export * from './layouts';
+export * from './internal';
```
Comment (dmiller9911) [inline packages/patternfly-4/react-core/src/index.ts]: kind of semantics, but if it is exported it is not really internals anymore. Maybe rename to `helpers` or something similar?

Comment (priley86) [reply packages/patternfly-4/react-core/src/index.ts]: i was thinking the same thing this morning actually... i'm up for this if others are.

Comment (jschuler) [reply packages/patternfly-4/react-core/src/index.ts]: helpers sounds good

Comment (priley86) [reply packages/patternfly-4/react-core/src/index.ts]: ok, works for me. Will update soon.

**Comment by tlabaj** [comment]
@priley86  looks like this one still needs some updates and conflicts resolved.

**Comment by priley86** [comment]
will rebase this and update to `helpers`soon after #1317 ... would be really helpful to have this.

File: packages/patternfly-4/react-core/src/helpers/util.ts [hunk 2]
```diff
@@ -13,11 +13,11 @@ export function getUniqueId(prefix = 'pf') {
   return `${prefix}-${uid}`;
 }
 
-export function debounce(func, wait) {
-  let timeout;
-  return (...args) => {
+export function debounce(this: any, func: (...args: any[]) => any, wait: number) {
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/helpers/util.ts]: Method signature has changed, is it needed to use `this` as first argument?

**Pull request merged**